### PR TITLE
e2e: serial tests fixes and tuning

### PR DESCRIPTION
**Pull request by fromanirh**
e2e tests improvement to support `scope=container` to complement #308

### CODE REVIEW

File: internal/noderesourcetopology/tostring.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		return ""
+	}
+	ret := ToString(resInfoList[0])
+	if len(resInfoList) == 1 {
+		return ret
+	}
+
+	var sb strings.Builder
+	fmt.Fprintf(&sb, "%s", ret)
+	for idx := 1; idx < len(resInfoList); idx++ {
```
[the remark this answers is not in the document]
Comment (fromanirh) [reply internal/noderesourcetopology/tostring.go:41]: This is to pretty print using the `strings.Builder`, to add a separator (being `,`) without introducing exceptional cases in the loop. Starting from 1 (and having 2+ elements ofc) we can always safely append `, item` (note leading comma-space).
Another alternative is to just use `string.Join`. I'm exploring alternatives here, but I don't mind reverting to `strings.Join` we find it more readable

[the remark this answers is not in the document]
Comment (Tal-or) [reply internal/noderesourcetopology/tostring.go:41]: Oh, I missed the comma-separator.
Thanks

[the remark this answers is not in the document]
Comment (fromanirh) [reply internal/noderesourcetopology/tostring.go]: unnecessarily hard to read anyway, changed

**Pull request merged**